Vagrantfile

```diff
@@ -0,0 +1,154 @@
+    # install Docker
+    if ! which docker; then
+      curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+      add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+      apt-get update
+      apt-get install -y docker-ce
+      groupadd docker
+      usermod -aG docker vagrant
+    fi
+
+    # install Docker Compose
+    if ! which docker-compose; then
+      mkdir -p /usr/local/bin
+      curl -L https://github.com/docker/compose/releases/download/1.23.2/docker-compose-`uname -s`-`uname -m` -o /usr/local/bin/docker-compose 2> /dev/null
+      chmod +x /usr/local/bin/docker-compose
+    fi
+
+    # install build prerequisites
+    sudo apt-get install -y make build-essential libssl-dev zlib1g-dev libbz2-dev \
+      libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+      xz-utils tk-dev libffi-dev liblzma-dev python-openssl git cowsay lolcat
+
+    # install nvm
+    if [ ! -e /opt/nvm ]; then
+      mkdir -p /opt/nvm
+      (curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.34.0/install.sh | NVM_DIR="/opt/nvm" bash) 2> /dev/null
+      chmod -R a+rwx /opt/nvm
+    fi
+    if [ ! -e /etc/profile.d/nvm.sh ]; then
+      cat > /etc/profile.d/nvm.sh #{HEREDOC} 'EOF'
+        #!/bin/sh
+        export NVM_DIR="/opt/nvm"
+      EOF
+      cat >> /home/vagrant/.bashrc #{HEREDOC} 'EOF'
+        # enable nvm
+        [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
+        [ -s "$NVM_DIR/bash_completion" ] && . "$NVM_DIR/bash_completion"
+      EOF
+    fi
+
+    # install pyenv
+    if [ ! -e /opt/pyenv ]; then
+      mkdir -p /opt/pyenv
+      git clone https://github.com/pyenv/pyenv.git /opt/pyenv
+      chmod -R a+rwx /opt/pyenv
+    fi
+    if [ ! -e /etc/profile.d/pyenv.sh ]; then
+      cat > /etc/profile.d/pyenv.sh #{HEREDOC} 'EOF'
+        #!/bin/sh
+        export PYENV_ROOT="/opt/pyenv"
+        export PATH="$PYENV_ROOT/bin:$PATH"
+      EOF
+      cat >> /home/vagrant/.bashrc #{HEREDOC} 'EOF'
+        # enable pyenv shell
+        if command -v pyenv 1>/dev/null 2>&1; then
+          eval "$(pyenv init -)"
+        fi
+      EOF
+    fi
+
+    # get the digitalmarketplace runner
+    sudo -H -u vagrant bash #{HEREDOC} EOF
+      if [ ! -e ~/digitalmarketplace ]; then
+        mkdir ~/digitalmarketplace
+        git clone https://github.com/alphagov/digitalmarketplace-runner ~/digitalmarketplace/digitalmarketplace-runner
+      fi
+    EOF
+
+    # install python, node, and yarn
+    sudo -H -u vagrant bash #{HEREDOC} EOF
+      cd ~/digitalmarketplace/digitalmarketplace-runner
+      source /etc/profile
+      pyenv install 3.6.8 2> /dev/null
+      pyenv global 3.6.8 system
+      source /opt/nvm/nvm.sh 2> /dev/null
+      nvm install 2> /dev/null
+      npm install -g yarn
+    EOF
+
+    # install system dependencies for communicating with postgres
+    apt-get install -y libpq-dev
```

Comment:
Can you tell me why this doesn't need sudo, and why you don't install it on l.81? I have no idea if it's right or wrong, so I'm curious to learn

Reply:
Woops, Line 81 doesn't actually need sudo; Vagrant provisioning scripts are already run as root.

As for why I don't install it with the rest, it's because I just felt it was neater; the other dependencies (apart from cowsay) where part of the build prerequisites needed for pyenv and nvm, whereas these dependencies are for their own special thing.